pkg/lca/seedgenerator.go

```diff
@@ -0,0 +1,294 @@
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the seedgenerator %s",
+		builder.Definition.Name)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+		if err == nil {
+			builder.Object = builder.Definition
+		}
+	}
+
+	return builder, err
+}
+
+// PullSeedGenerator pulls existing seedgenerator from cluster.
+func PullSeedGenerator(apiClient *clients.Settings, name string) (*SeedGeneratorBuilder, error) {
+	glog.V(100).Infof("Pulling existing seedgenerator name %s from cluster", name)
+
+	builder := SeedGeneratorBuilder{
+		apiClient: apiClient,
+		Definition: &lcasgv1alpha1.SeedGenerator{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the seedgenerator is empty")
+
+		builder.errorMsg = "seedgenerator 'name' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("seedgenerator object %s doesn't exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Delete removes the existing seedgenerator from a cluster.
+func (builder *SeedGeneratorBuilder) Delete() (*SeedGeneratorBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting the seedgenerator %s",
+		builder.Definition.Name)
+
+	if !builder.Exists() {
+		return builder, fmt.Errorf("seedgenerator cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.Delete(context.TODO(), builder.Definition)
+
+	if err != nil {
+		return builder, fmt.Errorf("can not delete seedgenerator: %w", err)
+	}
+
+	builder.Object = nil
+
+	return builder, nil
+}
+
+// Get returns seedgenerator object if found.
+func (builder *SeedGeneratorBuilder) Get() (*lcasgv1alpha1.SeedGenerator, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting seedgenerator %s",
+		builder.Definition.Name)
+
+	seedgenerator := &lcasgv1alpha1.SeedGenerator{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, seedgenerator)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return seedgenerator, err
+}
+
+// Exists checks whether the given seedgenerator exists.
+func (builder *SeedGeneratorBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if seedgenerator %s exists",
+		builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// WithSeedImage sets the seed image used by the seedgenerator.
+func (builder *SeedGeneratorBuilder) WithSeedImage(
+	seedImage string) *SeedGeneratorBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting seed image %s in seedgenerator", seedImage)
+
+	builder.Definition.Spec.SeedImage = seedImage
```

Comment:
It can be empty?

Reply:
yes - you can create a seedimage with empty spec.

`oc get seedgenerators.lca.openshift.io seedimage -o yaml
apiVersion: lca.openshift.io/v1alpha1
kind: SeedGenerator
metadata:
  creationTimestamp: "2024-01-18T17:05:02Z"
  generation: 1
  name: seedimage
  resourceVersion: "1896527"
  uid: 43fc7e08-68a6-42b2-b97e-64696aec2326
spec: {}
status:
  conditions:
  - lastTransitionTime: "2024-01-18T17:05:02Z"
    message: Seed Generation in progress
    observedGeneration: 1
    reason: InProgress
    status: "True"
    type: SeedGenInProgress
  observedGeneration: 1
`

Reply:
Not empty spec, it should be empty spec.seedImage